packages/edit-site/src/components/editor/index.js

```diff
@@ -165,6 +166,7 @@ export default function Editor() {
 					<GlobalStylesProvider>
 						<BlockContextProvider value={ blockContext }>
 							<SidebarComplementaryAreaFills />
+							<StartTemplateOptions />
```

Comment:
Is this the right pattern to show this modal:

 1- Right now, when you create a template, you have a first modal where you pick if it's a custom template, give it a name... 
 2- the template is created and you're redirected to the template page
 3- you're provided with options as starting points.

And you actually notice that a lot is happening at the same time when loading the template in the editor. I think it might be better if the "starting point" selection is part of the initial modal to "pre-configure" the template to create instead of being a post process. If we want to keep as it's own step, that modal could be multi-step (a wizard) if needed.

Reply:
We had a design that encapsulated the pre-configuration in a single modal, but feedback from Matias was to reuse what we have in the page creation flow (https://github.com/WordPress/gutenberg/pull/40034) to begin with, and refine holistically later.

Reply:
Exactly we are replicating the page creation flow. It also has some advantages if a template is empty when the user edits it, the modal appears even if it was already created before.